src/osp/OneStepProverHostIo.sol

```diff
@@ -31,6 +32,56 @@ contract OneStepProverHostIo is IOneStepProver {
     uint64 private constant INBOX_HEADER_LEN = 40;
     uint64 private constant DELAYED_HEADER_LEN = 112 + 1;
 
+    using BN254 for uint256[4];
+    using BN254 for uint256[3];
+    using BN254 for uint256[12];
+
+    // AlphaG1
+    // This is from the SRS points being used.
+    // This is the point at index 1, since index 0 is the generator value of the G1 group.
+    BN254.G1Point private ALPHA_G1 = BN254.G1Point(5421624913032980671919055010798735843841011930764711817607050648427876929258, 12995821280260994872112541311010834261076556242291585164372488699033268245381);
+
+    function computeGamma(uint256 z, uint256 y, uint256[2] memory p, uint256[4] memory alpha_minus_z_g2) internal pure returns (uint256) {
+        // Encode the variables and compute the keccak256 hash
+        bytes32 hash = keccak256(abi.encodePacked(z, y, p[0], p[1], alpha_minus_z_g2[0], alpha_minus_z_g2[1], alpha_minus_z_g2[2], alpha_minus_z_g2[3]));
+        return uint256(hash) % BN254.FR_MODULUS;
+    }
+
+    //  e((P - y) + gamma . (alpha - z), G2) = e((Q + gamma), (alpha - z)) 
```

Comment:
probably note, the last alpha - z is a passed in

Reply:
added.